### ISSUE TITLE
Create Page class

### ISSUE THREAD

**Issue by vladimino** [MagedAhmad/PHP-microframework#27]
# Context

Now it's time to take a look into the letter `V` of your `MVC` app. The way it is done now via the global function `view()` is going against the chosen OOP approach of the application design. Let's improve it!

# Proposal

1. _Create a class `Page` and move `view()` inside it_

This is definitely a responsibility of some class, not a global function.

2. _This time, don't use direct `require` when processing the template file_

The current behavior is not expected, there's no control over the output. A method within `Page` class should return the rendered content as a string and let the clients of `Page` decide when and how they want to output the result.

3. _Find a place[-s] to inject this new class to be able to show the template to the end-user_

Think of best practices of MVC pattern and call `Page->view()` from proper places (no hints this time). Don't forget about 404!

# To-do

To summarize, this issue will be considered done when all these subtasks are solved:

- [x] There's new class `Page` that is responsible for accepting template name and variables, processing the template and returning content as a string
- [ ] This class is injected in all appropriate places of the app and all calls to `view()` are updated to function as before
- [x] `functions.php` is removed (you don't use `dd()` in the code anyway, and this could be a separate helper class if you wish)

**Comment by MagedAhmad** [comment]
> Find a place[-s] to inject this new class to be able to show the template to the end-user

Now i use it in two places `HomeController@home` + `App@run` to render errors
where to inject it ? why i don't just `(new Page)->view` ? maybe later when i had more controllers i will create parent `Controller` where i inject the Page class and all controllers will extend that `Controller.php` .
am i thinking right ?

**Comment by vladimino** [comment]
> > Find a place[-s] to inject this new class to be able to show the template to the end-user
> 
> Now i use it in two places `HomeController@home` + `App@run` to render errors
> where to inject it ? why i don't just `(new Page)->view` ? maybe later when i had more controllers i will create parent `Controller` where i inject the Page class and all controllers will extend that `Controller.php` .
> am i thinking right ?

You already have a parent controller (in #29). Please inject`Page` class into it, it's essential for every controller.
